src/include/brain/brain_defs.h

```diff
@@ -5,36 +5,136 @@ namespace terrier::brain {
 enum class ExecutionOperatingUnitType : uint32_t {
   INVALID,
 
+  /**
+   * AGGREGATE_BUILD
+   * num_rows: # input rows to aggregation
+   * cardinality: # unique values (although probably more correct is # of unique values based on group by clause)
+   */
   AGGREGATE_BUILD,
+  /**
+   * AGGREGATE_ITERATE
+   * num_rows: # rows output by aggregation  // TODO(WAN): when is this not 1?
+   * cardinality: # unique values output (either 1 for non-group by or # unique values based on group by)
+   */
   AGGREGATE_ITERATE,
 
+  /**
+   * HASHJOIN_BUILD
+   * num_rows: # input rows
+   * cardinality: # unique tuples
+   */
   HASHJOIN_BUILD,
+  /**
+   * HASHJOIN_PROBE
+   * num_rows: number of probes
+   * cardinality: number of matches
+   */
   HASHJOIN_PROBE,
 
+  /**
+   * IDXJOIN
+   * num_rows: N/A
+   * cardinality: N/A
+   * This feature doesn't actually exist.
+   * The recorder emits a IDX_SCAN (with num_loops = # rows output by the outer loop)
+   */
   IDXJOIN,
```

Comment:
Do we still need this if the feature doesn't actually exist? Also, it seems that we haven't modified the `index_join_translator` to add the counters, especially for the `num_loops` feature?

Reply:
Removed.